plugins/modules/k8s_log.py

```diff
@@ -193,20 +201,28 @@ def execute_module(svc, params):
         resource = v1_pods
 
     kwargs = {}
-    if params.get("container"):
-        kwargs["query_params"] = {"container": params["container"]}
+    if module.params.get("container"):
+        kwargs["query_params"] = {"container": module.params["container"]}
+
+    if module.params.get("since_seconds"):
+        kwargs.setdefault("query_params", {}).update(
+            {"sinceSeconds": module.params["since_seconds"]}
+        )
 
-    if params.get("since_seconds"):
+    if module.params.get("previous"):
         kwargs.setdefault("query_params", {}).update(
-            {"sinceSeconds": params["since_seconds"]}
+            {"previous": module.params["previous"]}
         )
 
-    if params.get("previous"):
-        kwargs.setdefault("query_params", {}).update({"previous": params["previous"]})
+    try:
+        response = resource.log.get(
+            name=name, namespace=namespace, serialize=False, **kwargs
+        )
+    except ApiException as exc:
+        if exc.reason == "Not Found":
+            module.fail_json(msg="Pod {0}/{1} not found.".format(namespace, name))
```

Comment:
Rather than calling `module.fail_json()` here, can you instead raise a `CoreException` with the message? This will be caught and handled in `main()` in a way that's consistent with the rest of the module.